### ISSUE TITLE
Factor in `pointer-events: none` on visibility algorithm

### ISSUE THREAD

**Issue by HendrikThePendric** [cypress-io/cypress#9227]
### Current behavior
If an element is visible but has `pointer-events: none;` the `.should('be.visible')` fails, even though the element is in fact visible.

### Desired behavior

Should assert visibility only, which is distinct from "clickability". `should('be.visible')` should pass if the element is visible and should not take into account whether or not the element can be interacted with.

### Test code to reproduce

`index.html`
```html
<!DOCTYPE html>

<html lang="en">
    <head>
        <style>
            .overlay {
                position: fixed;
                top: 0;
                left: 0;
                width: 100vw;
                height: 100vh;
                background-color: red;
            }
            .tooltip-default {
                position: absolute;
                top: 100px;
                left: 100px;
                background-color: blue;
            }
            .tooltip-no-pointer-events {
                position: absolute;
                top: 300px;
                left: 300px;
                pointer-events: none;
                background-color: green;
            }
        </style>
    </head>

    <body>
        <p>Some content underneath</p>
        <div class="overlay">
            <div class="tooltip-default">Visible</div>
            <div class="tooltip-no-pointer-events">Also visible</div>
        </div>
    </body>
</html>
```
`index.js`
```js
describe('is visible issue', () => {
    it('asserts both tooltips are visible', () => {
        cy.visit('./index.html')

        cy.get('.tooltip-default').should('be.visible')

        /*  
        This assertion below fails with the following error:
        Timed out retrying: expected '<div.tooltip-no-pointer-events>' to be 'visible'

        This element <div.tooltip-no-pointer-events> is not visible because it has CSS property: position: fixed and it's being covered by another element
        */
        cy.get('.tooltip-no-pointer-events').should('be.visible')
    })
})
```
### Versions

`cypress@4.7.0`

### Additional note

This was already reported in of https://github.com/cypress-io/cypress/issues/698, see https://github.com/cypress-io/cypress/issues/698#issuecomment-591828047

**Comment by jennifer-shehane** [comment]
@HendrikThePendric Thanks for taking the time to provide a reproducible example. 

When opening an issue, make sure to always update to the [current version of Cypress](https://on.cypress.io/changelog) as your issue may have already been fixed. This is the case in your example. I think this was likely fixed in 4.12.0 in this PR https://github.com/cypress-io/cypress/pull/8095 but I didn't verify that exactly.

### 4.7.0

<img width="961" alt="Screen Shot 2020-11-18 at 1 03 09 PM" src="https://user-images.githubusercontent.com/1271364/99494074-c153d380-299e-11eb-8ffc-c87885dc3fc3.png">


### 5.6.0

<img width="1131" alt="Screen Shot 2020-11-18 at 1 03 44 PM" src="https://user-images.githubusercontent.com/1271364/99494050-ba2cc580-299e-11eb-9bb0-edcdea3ce775.png">

**Comment by andreasvirkus** [comment]
Should the docs be updated to reflect this change?
https://docs.cypress.io/guides/core-concepts/interacting-with-elements#Actionability
